File: UnitTests/ConvertInfixToPostFix.cs
using System;
using System.Collections.Generic;
using Calculator;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace UnitTests
{
    [TestClass]
    public class ConvertInfixToPostFix
    {
        [TestMethod]
        public void TestMethod1()
        {
            var infix = new List<object>() { 3.1, '+', 4.25 }; ;
            var postfix = new List<object>() {3.1,4.25,'+' };
            CollectionAssert.AreEqual(postfix, Core.ConvertInfixToPostfix(infix));
        }
        [TestMethod]
        public void TestMethod2()
        {
            var infix = new List<object>() { 9.0, '-', 4.6 };
            var postfix = new List<object>() { 9.0, 4.6, '-' };
            CollectionAssert.AreEqual(postfix, Core.ConvertInfixToPostfix(infix));
        }
        [TestMethod]
        public void TestMethod3()
        {
            var infix = new List<object>() { 4.0, '-', 9d };
            var postfix = new List<object>() { 4.0, 9d, '-' };
            CollectionAssert.AreEqual(postfix, Core.ConvertInfixToPostfix(infix));
        }
        [TestMethod]
        public void TestMethod4()
        {
            var infix = new List<object>() { 0.5, '*', 0.25 };
            var postfix = new List<object>() { 0.5, 0.25, '*' };
            CollectionAssert.AreEqual(postfix, Core.ConvertInfixToPostfix(infix));
        }
        [TestMethod]
        public void TestMethod5()
        {
            var infix = new List<object>() { 3d, '+', 4d, '*', 5d };
            var postfix = new List<object>() { 3d, 4d, 5d, '*', '+' };
            CollectionAssert.AreEqual(postfix, Core.ConvertInfixToPostfix(infix));
        }
        [TestMethod]
        public void TestMethod6()
        {
            var infix = new List<object>() { 3d, '*', 4d, '+', 5d };
            var postfix = new List<object>() { 3d, 4d, '*', 5d, '+' };
            CollectionAssert.AreEqual(postfix, Core.ConvertInfixToPostfix(infix));
        }
        [TestMethod]
        public void TestMethod7()
        {
            var infix = new List<object>() { '(', 3d, '+', 4d, ')', '*',  5d };
            var postfix = new List<object>() { 3d, 4d, '+', 5d, '*' };
            CollectionAssert.AreEqual(postfix, Core.ConvertInfixToPostfix(infix));
        }
        [TestMethod]
        public void TestMethod8()
        {
            var infix = new List<object>() { '(', 3d, '+', 4d,')', '*', '(', 5d, '+', 6d, ')' };
            var postfix = new List<object>() { 3d,4d, '+',5d,6d,'+','*' };
            CollectionAssert.AreEqual(postfix, Core.ConvertInfixToPostfix(infix));
        }
        [TestMethod]
        public void TestMethod9()
        {
            var infix = new List<object>() { 3d, '+', 4d, '+', 5d, '*', 6d, };
            var postfix = new List<object>() { 3d, 4d, '+', 5d, 6d, '*', '+' };
            CollectionAssert.AreEqual(postfix, Core.ConvertInfixToPostfix(infix));
        }
        [TestMethod]
        public void TestMethod10()
        {
            var infix = new List<object>() { 3d, '/', 4d, '*', '(',5d, '-', 6d, ')' };
            var postfix = new List<object>() { 3d, 4d, '/', 5d, 6d, '-', '*' };
            CollectionAssert.AreEqual(postfix, Core.ConvertInfixToPostfix(infix));
        }
        [TestMethod]
        public void TestMethod11()
        {
            var infix = new List<object>() { 4d, '*', '(', 5d, '-', 6d,'*',7d, ')' };
            var postfix = new List<object>() { 4d, 5d, 6d,7d,'*','-','*' };
            CollectionAssert.AreEqual(postfix, Core.ConvertInfixToPostfix(infix));
        }
        [TestMethod]
        public void TestMethod12()
        {
            var infix = new List<object>() { 3d, '+',4d, '*', '(', 5d, '-', 6d, '*', 7d, ')' };
            var postfix = new List<object>() { 3d, 4d, 5d, 6d, 7d, '*', '-', '*','+' };
            CollectionAssert.AreEqual(postfix, Core.ConvertInfixToPostfix(infix));
        }
    }
}

File: Calculator VB/UnitTests/ConvertInfixToPostFix.vb
Imports Calculator

Namespace UnitTests

    <TestClass>
    Public Class ConvertInfixToPostFix

        <TestMethod>
        Public Sub TestMethod1()
            Dim infix = New List(Of Object)() From {3.1, "+"c, 4.25}
            Dim postfix = New List(Of Object)() From {3.1, 4.25, "+"c}
            CollectionAssert.AreEqual(postfix, Core.ConvertInfixToPostfix(infix))
        End Sub

        <TestMethod>
        Public Sub TestMethod2()
            Dim infix = New List(Of Object)() From {
            9.0R, "-"c, 4.6}
            Dim postfix = New List(Of Object)() From {9.0R, 4.6, "-"c}
            CollectionAssert.AreEqual(postfix, Core.ConvertInfixToPostfix(infix))
        End Sub

        <TestMethod>
        Public Sub TestMethod3()
            Dim infix = New List(Of Object)() From {
            4.0R, "-"c,
            9.0R
            }
            Dim postfix = New List(Of Object)() From {
            4.0R,
            9.0R, "-"c}
            CollectionAssert.AreEqual(postfix, Core.ConvertInfixToPostfix(infix))
        End Sub

        <TestMethod>
        Public Sub TestMethod4()
            Dim infix = New List(Of Object)() From {0.5, "*"c, 0.25}
            Dim postfix = New List(Of Object)() From {0.5, 0.25, "*"c}
            CollectionAssert.AreEqual(postfix, Core.ConvertInfixToPostfix(infix))
        End Sub

        <TestMethod>
        Public Sub TestMethod5()
            Dim infix = New List(Of Object)() From {
            3.0R, "+"c,
            4.0R, "*"c,
            5.0R
            }
            Dim postfix = New List(Of Object)() From {
            3.0R,
            4.0R,
            5.0R, "*"c, "+"c}
            CollectionAssert.AreEqual(postfix, Core.ConvertInfixToPostfix(infix))
        End Sub

        <TestMethod>
        Public Sub TestMethod6()
            Dim infix = New List(Of Object)() From {
            3.0R, "*"c,
            4.0R, "+"c,
            5.0R
            }
            Dim postfix = New List(Of Object)() From {
            3.0R,
            4.0R, "*"c,
            5.0R, "+"c}
            CollectionAssert.AreEqual(postfix, Core.ConvertInfixToPostfix(infix))
        End Sub

        <TestMethod>
        Public Sub TestMethod7()
            Dim infix = New List(Of Object)() From {"("c,
            3.0R, "+"c,
            4.0R, ")"c, "*"c,
            5.0R
            }
            Dim postfix = New List(Of Object)() From {
            3.0R,
            4.0R, "+"c,
            5.0R, "*"c}
            CollectionAssert.AreEqual(postfix, Core.ConvertInfixToPostfix(infix))
        End Sub

        <TestMethod>
        Public Sub TestMethod8()
            Dim infix = New List(Of Object)() From {"("c,
            3.0R, "+"c,
            4.0R, ")"c, "*"c, "("c,
            5.0R, "+"c,
            6.0R, ")"c}
            Dim postfix = New List(Of Object)() From {
            3.0R,
            4.0R, "+"c,
            5.0R,
            6.0R, "+"c, "*"c}
            CollectionAssert.AreEqual(postfix, Core.ConvertInfixToPostfix(infix))
        End Sub

        <TestMethod>
        Public Sub TestMethod9()
            Dim infix = New List(Of Object)() From {
            3.0R, "+"c,
            4.0R, "+"c,
            5.0R, "*"c,
            6.0R
            }
            Dim postfix = New List(Of Object)() From {
            3.0R,
            4.0R, "+"c,
            5.0R,
            6.0R, "*"c, "+"c}
            CollectionAssert.AreEqual(postfix, Core.ConvertInfixToPostfix(infix))
        End Sub

        <TestMethod>
        Public Sub TestMethod10()
            Dim infix = New List(Of Object)() From {
            3.0R, "/"c,
            4.0R, "*"c, "("c,
            5.0R, "-"c,
            6.0R, ")"c}
            Dim postfix = New List(Of Object)() From {
            3.0R,
            4.0R, "/"c,
            5.0R,
            6.0R, "-"c, "*"c}
            CollectionAssert.AreEqual(postfix, Core.ConvertInfixToPostfix(infix))
        End Sub

        <TestMethod>
        Public Sub TestMethod11()
            Dim infix = New List(Of Object)() From {
            4.0R, "*"c, "("c,
            5.0R, "-"c,
            6.0R, "*"c,
            7.0R, ")"c}
            Dim postfix = New List(Of Object)() From {
            4.0R,
            5.0R,
            6.0R,
            7.0R, "*"c, "-"c, "*"c}
            CollectionAssert.AreEqual(postfix, Core.ConvertInfixToPostfix(infix))
        End Sub

        <TestMethod>
        Public Sub TestMethod12()
            Dim infix = New List(Of Object)() From {
            3.0R, "+"c,
            4.0R, "*"c, "("c,
            5.0R, "-"c,
            6.0R, "*"c,
            7.0R, ")"c}
            Dim postfix = New List(Of Object)() From {
            3.0R,
            4.0R,
            5.0R,
            6.0R,
            7.0R, "*"c, "-"c, "*"c, "+"c}
            CollectionAssert.AreEqual(postfix, Core.ConvertInfixToPostfix(infix))
        End Sub
    End Class
End Namespace